cmd/agent/daemon/enrichment/file_hash_enricher.go

```diff
@@ -50,56 +51,53 @@ type fileHashEnricher struct {
 	cache                  freelru.Cache[fileHashCacheKey, []byte]
 }
 
-func (enricher *fileHashEnricher) EventTypes() []castpb.EventType {
-	return []castpb.EventType{
-		castpb.EventType_EVENT_EXEC,
+func (enricher *fileHashEnricher) EventTypes() []events.ID {
+	return []events.ID{
+		events.SchedProcessExec,
 	}
 }
 
-func (enricher *fileHashEnricher) Enrich(ctx context.Context, req *EnrichRequest) {
-	e := req.Event
-	exec := e.GetExec()
-	if exec == nil || exec.Path == "" {
+func (enricher *fileHashEnricher) Enrich(ctx context.Context, in *types.Event, out *castpb.ContainerEvent) {
+	if in.Context.EventID != events.SchedProcessExec {
 		return
 	}
 
-	sha, err := enricher.calcFileHashForPID(req.EbpfEvent.Container, proc.PID(req.EbpfEvent.Context.NodeHostPid), exec.Path)
+	args, ok := in.Args.(types.SchedProcessExecArgs)
+	if !ok {
+		return
+	}
+
+	sha, err := enricher.calcFileHashForPID(in.Container, proc.PID(in.Context.NodeHostPid), args.Filepath)
 	if err != nil {
 		if errors.Is(err, ErrFileDoesNotExist) {
 			return
 		}
 	} else {
-		setExecFileHash(exec, sha)
+		out.GetExec().HashSha256 = sha
```

Comment:
Is this the same hash enrichment that is applied in the backend for exec events or this is a different kind of hash enrichment?